device/noise-protocol.go

```diff
@@ -16,14 +17,34 @@ import (
 	"golang.zx2c4.com/wireguard/tai64n"
 )
 
+type handshakeState int
+
+// TODO(crawshaw): add commentary describing each state and the transitions
```

Comment:
I've noticed you add these TODOs everywhere. Are these things that will actually be done at some point? If so, great, keep it. If not, let's get rid of this.